gammapy/utils/simulate.py

```diff
@@ -0,0 +1,79 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Simulate observations"""
+import numpy as np
+import astropy.units as u
+from ..cube import MapDataset, PSFKernel
+from ..cube import make_map_exposure_true_energy, make_map_background_irf
+from ..maps import WcsNDMap
+from ..cube.models import BackgroundModel
+
+__all__ = ["simulate_3d"]
+
+
+def simulate_3d(
+    skymodel, geom, pointing, irfs, livetime=1 * u.h, offset=0 * u.deg, edisp=False
+):
+
+    """Simulate a 3D dataset
+    
+    Simulate a source defined with a sky model for a given pointing,
+    geometry and irfs for a given exposure time.
+    This will return a dataset object which includes the counts cube,
+    the exposure cube, the psf cube, the background model and the sky model.
+
+    Parameters
+    ----------
+    skymodel : `~gammapy.cube.models.SkyModel`
+        Background model map
+    geom : `~gammapy.maps.wcs.WcsGeom`
+        Geometry object for the observation
+    pointing : `~astropy.coordinates.SkyCoord`
+        Pointing position
+    irfs : 
+        Irfs used for simulating the observation
+    livetime : `~astropy.units.quantity.Quantity`
+        Livetime exposure of the simulated observation
+    offset : `~astropy.units.quantity.Quantity`
+        Offset from the center of the pointing position.
+        This is used for the PSF and Edisp estimation
+    edisp : 
```

Comment:
state the explicit type. (boolean)

This maybe not needed if you instead rely on the presence or absence  of `edisp` in the `irf` dictionary.

Reply:
By default the irfs have an edisp so unless removing it by hand from the dict I don't see this being practical solution. 
My main idea behind having no edisp is that in a near future, including edisp might slow down a bit the model evaluation and fit so it's good to be able to have a simple simulation

Reply:
Here is a quick performance check using your `test_simulate` function and larger `WCSGeom` with 5 deg x 5 deg and a pixel size of 0.02:

<img width="596" alt="screenshot 2019-03-04 at 09 42 47" src="https://user-images.githubusercontent.com/3715928/53720928-32b5c080-3e62-11e9-8f7b-cff790577977.png">

As long as we have only as single energy dispersion for the whole map, it's definitely not the bottleneck in the model computation. So I think providing this option ist not needed and I'd prefer @registerrier suggestion to just include it, when it's present in the `irf` dict.

Reply:
ok, the difference for now is indeed quick small in simulation. I was thinking in terms of Fit() but I think it is also small for now with a simple Edisp matrix.